src/main/scala/medeia/generic/CoproductDerivationOptions.scala

```diff
@@ -0,0 +1,8 @@
+package medeia.generic
+
+case class CoproductDerivationOptions[A](
+    typeNameTransformation: PartialFunction[String, String] = PartialFunction.empty,
+    typeNameKey: String = "type"
+) {
+  val transformTypeNames: String => String = typeNameTransformation.orElse { case x => x }
```

Comment:
I think we could improve the naming here a bit :thinking: 

WDYT about `transform` only or `transformName` or `transformTypeName`?

Reply:
would opt for transformName if we keep the options separate and transformTypeName if we merge the two Option classes.

Reply:
:-1: for merging, `transformName` :+1: